trk/functions/index.js

```diff
@@ -65,4 +67,107 @@ exports.decrementUserTapCounter = functions.firestore
+            }
+
+            const tapData = tapDoc.data();
+            if (tapData.archived === undefined || tapData.archived === false) {
+                // Retrieve user's FCM token
+                const userRef = admin.firestore().collection('users').doc(tapData.user);
+                const userDoc = await userRef.get();
+
+                if (!userDoc.exists) {
+                    return res.status(404).send('User not found');
+                }
+                const userData = userDoc.data();
+                if (userData.fcmToken) {
+                    // Send a notification
+                    await sendNotificationToUser(userData.fcmToken, tapId, tapData.user);
+                    res.send(`Notification sent for tapId: ${tapId}`);
+                } else {
+                    res.status(500).send('FCM token not found for user');
+                }
+            } else {
+                res.send(`No notification sent for archived tapId: ${tapId}`);
+            }
+        } catch (error) {
+            console.error('Error:', error);
+            res.status(500).send('Error processing request');
+        }
+    });
+
+    // GCP Job (Queued), to call the notification function after 6 hours (when a session is started ONLY), on deletion of the starting tap of the session, we shift to the next most recent tap (scheduled for the same time as original with difference)
+    exports.scheduleFunction = functions.https.onCall(async (data, context) => {
+        const projectId = 'trk-app-505a1';
+        const queue = 'sessionNotifications';
+        const location = 'us-central1'; // e.g., 'us-central1'
+        const url = 'https://us-central1-trk-app-505a1.cloudfunctions.net/sessionNotificationFunction';
+        const payload = {tapId: data.tapId};
+        // Construct the fully qualified queue name.
+        const parent = client.queuePath(projectId, location, queue);
+
+        // Convert payload to base64 encoded string.
+        const convertedPayload = Buffer.from(JSON.stringify(payload)).toString('base64');
+
+        const task = {
+            httpRequest: {
+                httpMethod: 'POST',
+                url: url,
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: convertedPayload,
+            },
+            scheduleTime: {
+                seconds: ((data.expiryTime !== null)? data.expiryTime: (Math.floor(Date.now() / 1000) + 6 * 60 * 60)) // Scheduled  for the Session Expiration Time
+            },
+        };
+
+        await client.createTask({parent, task});
+        console.log(`Task created: ${task.name}`);
+    });
+
+    //Notification styling and packaging
+    async function sendNotificationToUser(fcmToken, tapId, userId) {
+        const message = {
+            token: fcmToken,
+            notification: {
+                title: 'Check out your Session',
+                body: `View your recent session in Profile!`,
```

Comment:
These prompts are great, and they work for now. Note for later, we should come up with more exciting alternatives.
e.g. "Session breakdown now available!", or "Session breakdown now created!". Something like that